src/core/labeling/qgsvectorlayerlabeling.h

```diff
@@ -99,6 +100,17 @@ class CORE_EXPORT QgsAbstractVectorLayerLabeling
      */
     virtual bool requiresAdvancedEffects() const = 0;
 
+    /**
+     * Multiply opacity by \a opacityFactor.
+     *
+     * This method multiplies the opacity of the labeling elements (text, shadow, buffer etc.)
+     * by \a opacity effectively changing the opacity of the whole labeling elements.
+     *
+     * \since QGIS 3.32
+     */
+    virtual void multiplyOpacity( double opacityFactor ) = 0;
```

Comment:
Making this pure virtual is an api break -- I'd go with:

```suggestion
    virtual void multiplyOpacity( double opacityFactor ) { Q_UNUSED( opacityFactor ); };
```